File: CPP/EASY/shuffleArray.cpp
//
// Created by yonahgraphics on 2021/07/14.
//
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {

        vector<int> intVec;

        //Loop from the beginning to n (the middle)
        for(int i = 0; i < n; i++)
        {
            intVec.push_back(nums[i]);
            intVec.push_back(nums.at(n+i));
        }

        return intVec;
    }
};

File: PYTHON/EASY/shuffleArray.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        # Index x and y values
        x = nums[:n]
        y = nums[n:]
        pair = []

        # Loop through the values of x and y and rearrange
        for x1, y1 in zip(x, y):
            pair.append(x1)
            pair.append(y1)

        return pair
